cmd/notify-mailer/main.go

```diff
@@ -233,42 +240,99 @@ func emailsForReg(id int, dbMap dbSelector) ([]string, error) {
 	return addresses, nil
 }
 
+type recipient struct {
+	id    int
+	Extra map[string]string
+}
+
+// emailToRecipientMap maps from an email address to a list of recipients with
+// that email address.
+type emailToRecipientMap map[string][]recipient
+
+// readRecipientsList reads a CSV filename and parses that file into a list of
+// recipient structs. It puts any columns after the first into a per-recipient
+// map from column name -> value.
+func readRecipientsList(filename string) ([]recipient, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	reader := csv.NewReader(f)
+	record, err := reader.Read()
+	if err != nil {
+		return nil, err
+	}
+	if record[0] != "id" {
```

Comment:
Can you add a check that `len(record) >= 1` before this? Probably overly paranoid but I think the mailer deserves it if anything does.